samples/get_vm_storage_policy.py

```diff
@@ -0,0 +1,276 @@
+        the specified entities
+    :rtype: pbm.profile.Profile[]
+    """
+
+    profiles = []
+    profileIds = profileManager.PbmQueryAssociatedProfile(ref)
+    if len(profileIds) > 0:
+        profiles = profileManager.PbmRetrieveContent(profileIds=profileIds)
+        return profiles
+    return profiles
+
+
+def ShowStorageProfileCapabilities(capabilities):
+    """Print vmware storage policy profile capabilities
+
+    :param capabilities: A list of VMware Storage Policy profile
+        associated capabilities
+    :type capabilities: pbm.capability.AssociatedPolicyCapabilities
+    :returns: None
+    """
+
+    for capability in capabilities:
+        for constraint in capability.constraint:
+            if hasattr(constraint, 'propertyInstance'):
+                for propertyInstance in constraint.propertyInstance:
+                    print("\tKey: {} Value: {}".format(propertyInstance.id,
+                                                       propertyInstance.value))
+
+
+def ShowStorageProfile(profiles):
+    """Print vmware storage policy profile
+
+    :param profiles: A list of VMware Storage Policy profiles
+    :type profiles: pbm.profile.Profile[]
+    :returns: None
+    """
+
+    for profile in profiles:
+        print("Name: {}{}{} ".format(bcolors.OKGREEN,
+                                     profile.name,
+                                     bcolors.ENDC))
+        print("ID: {} ".format(profile.profileId.uniqueId))
+        print("Description: {} ".format(profile.description))
+        if hasattr(profile.constraints, 'subProfiles'):
+            subprofiles = profile.constraints.subProfiles
+            for subprofile in subprofiles:
+                print("RuleSetName: {} ".format(subprofile.name))
+                capabilities = subprofile.capability
+                ShowStorageProfileCapabilities(capabilities)
+
+
+def SearchVMByName(serviceInstance, name, strict=False):
+    """Search virtual machine by name
+
+    :param serviceInstance: A ServiceInstance managed object
+    :type name: serviceInstance
+    :param name: A virtual machine name
+    :type name: str
+    :param strict: A flag used to set strict search method
+        (default is False)
+    :type strict: bool
+    :returns: A virtual machine object
+    :rtype: VirtualMachine
+    """
+
+    content = serviceInstance.content
+    root_folder = content.rootFolder
+    objView = content.viewManager.CreateContainerView(root_folder,
+                                                      [vim.VirtualMachine],
+                                                      True)
+    vmList = objView.view
+    objView.Destroy()
+    obj = []
+    for vm in vmList:
+        if strict:
+            if (vm.name == name):
+                obj.append(vm)
+                return obj
+        else:
+            if re.match(".*{}.*".format(name), vm.name):
+                obj.append(vm)
+    return obj
+
+
+def GetArgs():
```

Comment:
Please move this to top of the file to be in consistent with other  sample files

Reply:
Changes have also been made in samples/set_vm_storage_policy.py